src/main.rs

```diff
@@ -118,16 +141,30 @@ fn sync_list(path: &str, pattern: &str, list_id: &str) -> Result<(), Box<std::er
             .collect()
     });
 
-    let list_name = "to-watch";
-    let result = film_ids.and_then(|ids| {
-        let mut request = letterboxd::ListUpdateRequest::new(String::from(list_name));
-        request.entries = ids.into_iter()
-            .map(letterboxd::ListUpdateEntry::new)
-            .collect();
-        client.patch_list(list_id, &request, &token)
+    // Fetch ids for films already on list.
+    // TODO: The request just fetches the first X items not all.
```

Comment:
Do you want to resolve it now? Or next PR?

Reply:
This could be a solution
```rust
/// Return film ids of all films in the list
fn saved_film_ids<'a>(
    list_id: &'a str,
    client: &'a letterboxd::Client,
    token: &'a letterboxd::AccessToken,
) -> impl future::Future<Item = HashSet<String>, Error = letterboxd::Error> + 'a {
    let init_state: (Option<String>, _) = (None, HashSet::new());
    future::loop_fn(init_state, move |state| {
        let mut req = letterboxd::ListEntriesRequest::default();
        let (cursor, film_ids) = state;
        req.cursor = cursor;
        client.list_entries(list_id, &req, Some(token)).and_then(
            move |resp| {
                let mut film_ids = film_ids;
                for list_entry in resp.items.into_iter() {
                    film_ids.insert(list_entry.film.id);
                }
                if resp.next.is_some() {
                    Ok(future::Loop::Continue((resp.next, film_ids)))
                } else {
                    Ok(future::Loop::Break(film_ids))
                }
            },
        )
    })
}
```

Reply:
That's an interesting approach. Let's do it in another PR.